src/tools/perf/perftest_daemon.c

```diff
@@ -347,15 +348,48 @@ ucp_perf_daemon_send_handler(void *arg, const void *header,
     return UCS_OK;
 }
 
+static void ucp_perf_daemon_cleanup_connections(ucp_perf_daemon_context_t *ctx)
+{
+    ucs_trace("cleaning up daemon connections");
+
+    if (NULL != ctx->send_memh) {
+        /* coverity[check_return] */
+        ucp_mem_unmap(ctx->context, ctx->send_memh);
+        ctx->send_memh = NULL;
+    }
+
+    if (NULL != ctx->recv_memh) {
+        /* coverity[check_return] */
+        ucp_mem_unmap(ctx->context, ctx->recv_memh);
+        ctx->recv_memh = NULL;
+    }
+
+    ucp_perf_daemon_ep_close(ctx, ctx->peer_ep);
+    ctx->peer_ep = NULL;
+
+    ucp_perf_daemon_ep_close(ctx, ctx->host_ep);
+    ctx->host_ep = NULL;
+}
+
 static ucs_status_t
 ucp_perf_daemon_fin_handler(void *arg, const void *header, size_t header_length,
                             void *data, size_t length,
                             const ucp_am_recv_param_t *param)
 {
+    uint8_t keep_running;
```

Comment:
maybe define struct type for the FIN message?

Reply:
ok